rm-config/src/keymap.rs

```diff
@@ -0,0 +1,425 @@
+use std::{
+    collections::HashMap, io::ErrorKind, marker::PhantomData, path::PathBuf, sync::OnceLock,
+};
+
+use anyhow::Result;
+use crossterm::event::{KeyCode, KeyModifiers};
```

Comment:
maybe rename `KeyModifiers` to `CrosstermKeyModifiers`? was a little confused since we have `KeyModifiers` and `KeyModifier` lol